README.md

```diff
@@ -17,149 +17,264 @@ Dead simple fluent API. Customizable. Reusable.
+validation process with exception based tests. And of course, perform
+asynchronous tests as well. All in one library.
 
-We can also use the `check` function to perform exception based validations.
-This is going to throw an exception when the validation fails:
+#### Boolean based validation:
 
 ```javascript
-import v8n from "v8n";
+v8n()
+  .string()
+  .first("H")
+  .test("Hello"); // true
+```
+
+#### Exception based validation:
 
+```javascript
 try {
   v8n()
-    .number()
-    .between(10, 20)
-    .check(25);
+    .string()
+    .first("b")
+    .check("foo");
 } catch (ex) {
-  // ex is a ValidationException object
+  console.log(ex.rule.name); // first
 }
 ```
 
-> A `ValidationException` object contains a lot of useful information about the
-validation process and its fail cause.
+#### Getting all failures:
+
+```javascript
+const failed = v8n()
+  .string()
+  .minLength(3)
+  .testAll(10);
+
+failed;
+// [
+//   ValidationException { rule: { name: "string", ... } },
+//   ValidationException { rule: { name: "minLength", ... } }
+// ]
+```
 
-### Asynchronous validation
+#### Async validation:
 
-If your validation strategy includes some asynchronous rule, you must use the
-'testAsync' function, so that the validation process will execute
-asynchronously. It will return a promise that will resolve with the validated
-value when it's valid and rejects with a `ValidationException` when it's
-invalid.
+If your validation strategy has some rule that performs time consuming
+validation, like a back-end check validation, you should use asynchronous
+validation:
 
 ```javascript
 v8n()
-  .number()
-  .between(10, 100)
-  .someAsyncRule()
-  .testAsync(50)
-  .then(value => { /* valid */ })
-  .catch(exception => { /* invalid */ });
+  .somAsyncRule()
+  .testAsync("foo") // returns a Promise
+  .then(result => {
+    /* valid! */
+  })
+  .catch(ex => {
+    /* invalid! */
+  });
+```
+
+### Shareable
+
+Share your rules with the world, and use theirs as well.
+
+Create useful validation rules and share them with the open source community,
+and let people around the world to validate without reinventing the wheel.
+
+### Ready to use!
+
+There are a lot of built-in rules and modifiers for you to use already
+implemented in `v8n`'s core. Take a look at all of them in our
+[API](https://imbrn.github.io/v8n/api/) page. But if you can't find what you
+need, go ahead and make it.
+
+### Tiny!
+
+All these incredible features for just a few bytes:
+
+![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/v8n.svg)
+
+## Installation
+
+```shell
+# Using npm
+npm install v8n
+
+# or yarn
+yarn add v8n
 ```
 
-## Rules
+### Or using a `script` tag:
+
+```html
+<!-- From unpkg -->
+<script src="https://unpkg.com/v8n/dist/v8n.min.js"></script>
+
+<!-- or from jsdelivr -->
+<script src="https://cdn.jsdelivr.net/npm/v8n/dist/v8n.min.js"></script>
+```
+
+## More about v8n
```

Comment:
Just link prominently to Getting Started, this seems like a duplicate with different text even. If anything make sure this and docs are in sync. imo not necessary on GitHub again.